src/main/java/it/gov/pagopa/pu/bff/service/debt_position/DebtPositionRetrieverServiceImpl.java

```diff
@@ -99,4 +112,35 @@ public boolean deleteDebtPosition(Long organizationId, Long debtPositionId, User
     return debtPositionService.deleteDebtPosition(debtPositionId,accessToken);
   }
 
+  @Override
+  public Resource getDebtPositionNoticesZip(Long organizationId, Long debtPositionId, UserInfo loggedUser, String accessToken) {
+    AuthorizationService.validateUserForOrganizationId(organizationId, loggedUser);
+
+    DebtPositionDTO debtPosition = debtPositionService.getDebtPosition(debtPositionId, accessToken);
+    if (debtPosition == null){
+      return null;
+    }
+
+    List<FileResourceDTO> pdfResources = debtPosition
+      .getPaymentOptions()
+      .stream()
+      .flatMap(po ->
+        po.getInstallments()
+          .stream()
+          .filter(
+            i ->
+              (InstallmentStatus.UNPAID.equals(i.getStatus()) ||
+                InstallmentStatus.UNPAYABLE.equals(i.getStatus())))
+      )
```

Comment:
Instead of retrieving the entire DP, and next filter to search unpaid Installment lets:
1. check if the user is authorized on DebtPositionTypeOrg (@MScarsella there is some API that we are implementing which, given a debtPositionId, could ask to debt-positions ms to verify if the current operator has grants on it?)
2. lets call directly an API exposed through SpringDataRest on InstallmentNoPII which would return the unpaid InstallmentNoPII (as done in findPaidByIun, but using it.gov.pagopa.pu.debtpositions.util.InstallmentUtils#INVALIDABLE_STATUSES constant instead)

Reply:
The printable statuses are a concern of just the BFF, there are not constants in debt-positions available for that.
let's expose a query on InstallmentNoPII repository having `debtPositionId` and `statuses` as input